nisystemlink/clients/file/_file_client.py

```diff
@@ -0,0 +1,177 @@
+        """Get information about available API operations.
+
+        Returns:
+            Information about available API operations.
+
+        Raises:
+            ApiException: if unable to communicate with the File Service.
+        """
+
+    @get(
+        "service-groups/Default/files",
+        args=[
+            Query,
+            Query,
+            Query(name="orderBy"),
+            Query(name="orderByDescending"),
+            Query(name="id"),
+        ],
+    )
+    def __get_files(
+        self,
+        skip: int = 0,
+        take: int = 0,
+        order_by: Optional[str] = None,
+        order_by_descending: Optional[str] = "false",
+        file_ids: Optional[str] = None,
+    ) -> models.FileQueryResponse:
+        """Lists available files on the SystemLink File service.
+        Use the skip and take parameters to return paged responses.
+        The orderBy and orderByDescending fields can be used to manage sorting the list by metadata objects.
+
+        Args:
+            skip: How many files to skip in the result when paging. Defaults to 0.
+            take: How many files to return in the result, or 0 to use a default defined by the service.
+              Defaults to 0.
+            order_by: The name of the metadata key to sort by. Defaults to None.
+            order_by_descending: The elements in the list are sorted ascending if "false"
+              and descending if "true". Defaults to "false".
+            file_ids: Comma-separated list of file IDs to search by. Defaults to None.
+
+        Returns:
+            File Query Response
+
+        Raises:
+            ApiException: if unable to communicate with the File Service.
+        """
+
+    def get_files(
+        self,
+        skip: int = 0,
+        take: int = 0,
+        order_by: Optional[models.FileQueryOrderBy] = None,
+        order_by_descending: Optional[bool] = False,
+        file_ids: Optional[str] = None,
+    ) -> models.FileQueryResponse:
+        """Lists available files on the SystemLink File service.
+        Use the skip and take parameters to return paged responses.
+        The orderBy and orderByDescending fields can be used to manage sorting the list by metadata objects.
+
+        Args:
+            skip: How many files to skip in the result when paging. Defaults to 0.
+            take: How many files to return in the result, or 0 to use a default defined by the service.
+            Defaults to 0.
+            order_by: The name of the metadata key to sort by. Defaults to None.
+            order_by_descending: The elements in the list are sorted ascending if False
+            and descending if True. Defaults to False.
+            file_ids: Comma-separated list of file IDs to search by. Defaults to None.
+
+        Returns:
+            File Query Response
+
+        Raises:
+            ApiException: if unable to communicate with the File Service.
+        """
+        # Uplink does not support enum serializing into str
```

Comment:
This is annoying huh? I'm ok with this workaround for now. I suspect there may be other workarounds, but I'm for getting this API in soon even if it needs to have this workaround.

Reply:
I don't know how to fix the bool serialization, though. Normally our services would be case insensitive for stuff like this.

Reply:
Yes, I saw your TODO note and went with the wrapper method instead of Literal to keep it simple for the user anyway, I need to workaround the case sensitivity for boolean for that API.